rust/parquet/src/reader/schema.rs

```diff
@@ -72,6 +73,18 @@ where
         .map(|fields| Schema::new(fields))
 }
 
+/// Convert parquet column schema to arrow field.
+pub fn parquet_to_arrow_field(parquet_column: ColumnDescPtr) -> Result<Field> {
+    let schema = parquet_column.self_type_ptr();
+
+    let mut leaves = HashSet::new();
+    leaves.insert(parquet_column.self_type() as *const Type);
```

Comment:
I'm curious why we need to deal with a pointer here?

Reply:
Since parquet library has no things such as type id. If we don't use pointer, we must use things like string, which is much less efficient than a pointer.